jhputnam/test_mailroom.py

```diff
@@ -0,0 +1,140 @@
+"""
+Mailroom testing.
+"""
+
+import pickle
+import pytest
+
+import mailroom
+
+
+def test_initdb_dbpresent():
+    """Test dbinit with db file present."""
+
+    testdb = {'Dan Rather': [3453.0], 'SpongeBob': [44.05]}
+
+    # Call initdb with test data.
+    donors = mailroom.initdb('testdb.dat')
+
+    # Assert that the test data is what we expect.
+    assert donors == testdb
+
+
+def test_initdb_dbmissing1(monkeypatch):
+    """Test dninit function with db missing and option 1."""
+
+    # Override input and supply option 1.
+    monkeypatch.setitem(__builtins__, 'input', lambda x: '1')
+
+    # Call initdb to assign donors.
+    donors = mailroom.initdb('testdb2.dat')
+
+    # Assert that the dict is empty.
+    assert donors == {}
+
+
+def test_initdb_dbmissing2(monkeypatch, capsys):
+    """Test dbinit function with db missing and option 2."""
+
+    # Override input and supply option 2.
+    monkeypatch.setitem(__builtins__, 'input', lambda x: '2')
+
+    # Catch the SystemExit.
+    with pytest.raises(SystemExit):
+        mailroom.initdb('testdb3.dat')
```

Comment:
Unittest has a built in for self.assertRaises(), but I think this is very interesting using a context manager here.